### PR TITLE
Make ReserveEntityIdsResponse contain an iterator

### LINKED ISSUE #65: ReserveEntityIdsResponse should return the list of entities as an iterator
Currently (as of my PR which implements send_component_update), `ReserveEntityIdsResponse` defines a range of entity IDs as: "first entity ID" and "number of entity IDs".

We can probably turn this into some kind of generator pattern, that allows someone to `.take()` an entity ID from the reserved range.

### PR DESCRIPTION
**Pull request by jamiebrynes7**
Reworked the `ReservedEntityIdsRange` to be an iterator and hid the internals. 

Fixes #65

### CODE REVIEW

File: spatialos-sdk/src/worker/op.rs [hunk 1]
```diff
@@ -482,8 +479,35 @@ pub struct ReserveEntityIdsResponseOp {
...
 #[derive(Debug)]
 pub struct ReservedEntityIdRange {
-    pub first_entity_id: EntityId,
-    pub number_of_entity_ids: u32,
+    current: i64,
+    consumed: u32,
+    reserved: u32,
+}
+
+impl ReservedEntityIdRange {
```
Comment (davedissian) [inline spatialos-sdk/src/worker/op.rs]: Why not instead use `std::ops::Range` and delete the custom type entirely?

Comment (jamiebrynes7) [reply spatialos-sdk/src/worker/op.rs]: Because I didn't know that existed 😂

Comment (jamiebrynes7) [reply spatialos-sdk/src/worker/op.rs]: It seems very appropriate - will refactor

Comment (jamiebrynes7) [reply spatialos-sdk/src/worker/op.rs]: Sadly we lose the `Iterator` impl as `EntityId` must implement [`Step`](https://doc.rust-lang.org/std/iter/trait.Step.html) to get the blanket implementation for `Range<EntityId>`.

Unfortunately, `Step` is nightly and I don't want to force us onto a nightly version

Comment (jamiebrynes7) [reply spatialos-sdk/src/worker/op.rs]: What I can do - is add a TODO to replace with `Range<EntityId>` once `Step` is stabilised.

**Pull request merged**